python/thunder/extraction/source.py

```diff
@@ -1,6 +1,9 @@
 from numpy import asarray, mean, sqrt, ndarray, amin, amax, concatenate, sum, zeros, maximum, \
     argmin, newaxis, ones, delete, NaN, inf, isnan
 
+import numpy.linalg as la
```

Comment:
Can we move this one so it's locally scoped to `distance`, and use `from numpy.linalg import norm`?